File: ruby/grade-school/grade_school.rb
class School
  attr_reader :students

  def initialize
    @students = Hash.new { |student, grade| student[grade] = [] }
  end

  def to_hash
    sorted = students.map { |grade, students| [grade, students.sort] }.sort
    Hash[sorted]
  end

  def add(name, grade)
    students[grade] << name
  end

  def grade(grade)
    students[grade].sort
  end

end

File: javascript/grade-school/grade-school.js
var School = function() {
  this.students = {}
};

School.prototype.roster = function(){
  for (var i = 0; i < Object.keys(this.students).length; i++) {
    this.students[Object.keys(this.students)[i]].sort()
  };
  return this.students;
}

School.prototype.add = function(name, grade){
  if(this.students[grade]){
    this.students[grade].push(name)
  }else{
    this.students[grade] = [name]
  };
};

School.prototype.grade = function(grade){
  if(!this.students[grade]) return [];
  return this.students[grade].sort();
};

module.exports = School;